field.cpp

```diff
@@ -0,0 +1,194 @@
+                        int merged = field[i][j].moveCellToDirection(direction);
+                        incrementForScore += merged;
+                    }
+                }
+                //std::cout << "incrementForScore: " << incrementForScore << std::endl;
+            }          
+        }
+        else if(direction == Direction::LEFT || direction == Direction::RIGHT){
+            for(int i = 0; i < size; i++){
+                for(int j = startPoints["start"]; j != startPoints["end"]  + startPoints["step"]; j += startPoints["step"]){
+                    // std::cout << "current i/j: " << i << ' ' << j << std::endl;
+                    // if(field[i][j].getNeighbor(direction) != nullptr){
+                    //     std::cout << "neighbor: " << field[i][j].getNeighbor(direction) << std::endl;
+                    // }else{
+                    //     std::cout << "neighbor: " << "nullptr" << std::endl;
+                    // }
+
+                    if(field[i][j].getFilledStatus()){
+                        int merged = field[i][j].moveCellToDirection(direction);
+                        incrementForScore += merged;
+                        //std::cout << "incrementForScore: " << incrementForScore << std::endl;
+                    }
+                    //std::cout<< "j++" << std::endl;
+                }
+                //std::cout << "incrementForScore: " << incrementForScore << std::endl;
+                //std::cout<< "i++" << std::endl;
+            }
+        }
+        return incrementForScore;
+    }
+    return incrementForScore;
+}
+
+void Field::printFieldConsole(){
+    for(int i = 0; i < size; i++){
+        for(int j = 0; j < size; j++){
+            std::cout << field[i][j].getCellValue() << " ";
+        }
+        std::cout << std::endl;
+    }
+}
+
+std::list<std::pair<int,int>> Field::findFreeCells(){
+    std::list<std::pair<int,int>> cordsOfFreeCells;
+    for(int i = 0; i < size; i++){
+        for(int j = 0; j < size; j++){
+            if(!field[i][j].getFilledStatus()){
+                cordsOfFreeCells.push_back({i,j});
+            }
+        }
+    }
+
+    return cordsOfFreeCells;
+}
+
+int Field::createRandomNummber(int min, int max){
+    std::random_device random;
+    std::mt19937 generator((random()));
+    std::uniform_int_distribution<> voter (min, max);
+
+    return voter(generator);
+}
+
+void Field::FillFreeCells(int numCellsToFill){
+    std::list<std::pair<int,int>> cordsOfFreeCells = findFreeCells();
+
+    if(numCellsToFill > cordsOfFreeCells.size()){
+        numCellsToFill = cordsOfFreeCells.size();
+    }
+
+    if (numCellsToFill == 0){
+        return;
+    }else if(numCellsToFill == 1 && cordsOfFreeCells.size() == 1){
+        field[cordsOfFreeCells.begin()->first][cordsOfFreeCells.begin()->second].setCellValue();
+        return;
+    }else{
+        for(int i = 0; i < numCellsToFill; i++){
+            int freeCellIndex = createRandomNummber(0, cordsOfFreeCells.size()-1);
```

Comment:
Why can't we just use setCellValue here also?

Reply:
setCellValue already has a random generator

Reply:
done